### PR TITLE
Magnum Shot

### PR DESCRIPTION
**Pull request by Tachiorz**
Magnum Shot skill. No splash damage

### CODE REVIEW

File: src/ChannelServer/Skills/Combat/MagnumShot.cs [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+				// Deal with it!
+				if (damage > 0)
+					target.TakeDamage(tAction.Damage = damage, attacker);
+
+				// TODO: we have to calculate knockback distance right
+				// TODO: target with defence up shouldn't be knocked back
+				attacker.Shove(target, KnockBackDistance);
+				
+				tAction.Set(TargetOptions.KnockDownFinish);
```
Comment (exectails) [inline src/ChannelServer/Skills/Combat/MagnumShot.cs]: You're setting KnockDownFinish when the target isn't dead? Is that correct? If so, that option includes Finished, so you don't have to set that one in the next if.

File: src/ChannelServer/Skills/Combat/MagnumShot.cs [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+				// is dead
+				if (target.IsDead)
+				{
+					aAction.Set(AttackerOptions.KnockBackHit1);
+					tAction.Set(TargetOptions.Finished);
+				}
+			}
+			else
+			{
+				aAction.Set((AttackerOptions)0x800);
```
Comment (exectails) [inline src/ChannelServer/Skills/Combat/MagnumShot.cs]: I guess if this one appears when you miss it's a "Missed" option. Feel free to add it to the enum.

File: src/ChannelServer/Skills/Combat/MagnumShot.cs [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+		/// <summary>
+		/// Returns the raw damage to be done.
+		/// </summary>
+		/// <param name="attacker"></param>
+		/// <param name="skill"></param>
+		/// <returns></returns>
+		protected float GetDamage(Creature attacker, Skill skill)
+		{
+			var result = attacker.GetRndTotalDamage();
```
Comment (exectails) [inline src/ChannelServer/Skills/Combat/MagnumShot.cs]: Is this correct? Shouldn't it be ranged damage? Total damage calculates the average damage between the right and left hand, under the assumption that you're using melee weapons.

File: src/ChannelServer/Skills/Combat/MagnumShot.cs (hunk 1 above)
Comment (Tachiorz) [inline src/ChannelServer/Skills/Combat/MagnumShot.cs]: not FinishingKnockDown includes Finished, not KnockDownFinish

File: src/ChannelServer/Skills/Combat/MagnumShot.cs (hunk 3 above)
Comment (Tachiorz) [inline src/ChannelServer/Skills/Combat/MagnumShot.cs]: oops, copypasted that from smash and messed up

File: src/ChannelServer/Skills/Combat/MagnumShot.cs (hunk 1 above)
Comment (exectails) [inline src/ChannelServer/Skills/Combat/MagnumShot.cs]: Oh, sorry. Misread. Stupid option name =|

**Pull request merged**